src/header.rs

```diff
@@ -615,10 +613,14 @@ impl Header {
     /// major device number.
     pub fn set_device_major(&mut self, major: u32) -> io::Result<()> {
         if let Some(ustar) = self.as_ustar_mut() {
-            return Ok(ustar.set_device_major(major));
+            return {
+                ustar.set_device_major(major);
+                Ok(())
+            };
```

Comment:
I don't think this change is necessary?

Reply:
Or otherwise returning a block feels overly complicated, could this be factored as two statements?

Reply:
done.